### ISSUE TITLE
Does not work on Mac M1

### ISSUE THREAD

**Issue by meyntony** [microsoft/mssql-docker#734]
```
docker run -e "ACCEPT_EULA=Y" -e "SA_PASSWORD=<YourStrongPassw0rd#4b4b5c9e-cec8-49c1-8b50-a59e47ba4c63>" -p 56038:1433 --name umbracocloud-local-dev-sql-Modular-56038 -h Modular.umbracocloud-local-dev-sql -d mcr.microsoft.com/mssql/server:2019-latest
docker exec -it umbracocloud-local-dev-sql-Modular-56038 /opt/mssql-tools/bin/sqlcmd -S localhost -U SA -P "<YourStrongPassw0rd#4b4b5c9e-cec8-49c1-8b50-a59e47ba4c63>" -Q "CREATE DATABASE LocalDevUmbracoCloudModular"
```

The above code works fine on Windows, Ubuntu, Mac with Intel Chip, but not on Mac with the M1 chip.

Is this a known issue?

**Comment by iOvergaard** [comment]
I am trying out the commands that @meyntony wrote on an M1 Pro (arm64) processor and getting the following warning when spinning up the container:

```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

I figure this explains why it would or could give problems. 

A possible workaround, [following this guide](https://database.guide/how-to-install-sql-server-on-an-m1-mac-arm64/), suggests to use the image `mcr.microsoft.com/azure-sql-edge` instead which seemingly works on the arm64 arch pretty much by changing the image name and adding some Linux capabilities:

```bash
docker run --cap-add SYS_PTRACE -e 'ACCEPT_EULA=1' -e 'MSSQL_SA_PASSWORD=P@ssw0rd' -p 1433:1433 --name sqledge -d mcr.microsoft.com/azure-sql-edge
```

In terms of running the `sqlcmd` command afterwards, I am not sure how to do that as it looks like the command resides in another place and I have not looked much into it.

Even so, I am able to connect to the database server using Azure Data Studio (which works on Apple Silicon even though it's not officially supported) and create new databases.

**Comment by rjschave** [comment]
@iOvergaard,

The sqlcmd utility is not included in the ARM64 version of the SQL Edge container.  I pulled another image that contains sqlcmd.

Retreive image w/ sqlcmd tools:
`docker pull mcr.microsoft.com/mssql-tools`

Assuming the  SQL Edge container is running on the same machine and bound to the default port of 1433, you could try something like this:
`docker run -it -d --platform linux/amd64 mcr.microsoft.com/mssql-tools /opt/mssql-tools/bin/sqlcmd -S host.docker.internal -U <user_name> -P '<password>' -Q 'SELECT * FROM a_table'`

**Comment by iOvergaard** [comment]
Nice one, @rjschave. Thanks for the info!

**Comment by roboticflamingo** [comment]
Please add aarch64 and arm64 support (necessary if one is using docker on M1 mac).

**Comment by sshquack** [comment]
More people are running [into this issue](https://hub.docker.com/_/microsoft-mssql-server?tab=reviews) and using azure sql edge as a workaround.
<img width="634" alt="Screenshot 000072@2x" src="https://user-images.githubusercontent.com/5678459/153539738-01b863da-e2f4-42a3-a60d-4c81fa234e29.png">

Please add support for ARM64 / Apple silicon / M1 to MS SQL Server docker image.

**Comment by eriklumme** [comment]
👍 'ed this issue. I'm using Azure SQL Edge as a workaround, but the lack of CLR support is an issue, as the database I'm working with uses the `FORMAT` function heavily.

**Comment by gnochi** [comment]
Since Azure SQL Edge does not support all the features of SQL Server its not an appropriate alternative for everyone. I would highly appreciate if Microsoft would support the M1 soon.

**Comment by Justinio99** [comment]
If there is no support for MSSQL we as a company will look into other solutions! To me thats a must have.

**Comment by mathibla** [comment]
M1 support is needed urgently

**Comment by Dihenry06** [comment]
M1 support is needed urgently

**Comment by nathan-palmer** [comment]
We will be transitioning our container-based DB work off of SQL Server if aarch64 / arm64 / M1 support does not happen soon.  Azure SQL Edge is not a suitable substitute.

**Comment by coolwuu** [comment]
M1 support is needed urgently

**Comment by coolwuu** [comment]
M1 support is needed urgently

**Comment by louisektw** [comment]
M1 support is needed urgently

**Comment by sirmahyar** [comment]
M1 support is needed urgently

**Comment by tracker1** [comment]
I would suggest reducing the "me too" comments and simply thumbs up on the primary post at the top.

I'm not working for MS, but would be surprised if this sees any attention in the near term.  Mostly because M1 isn't really a server target for MS SQL Server.  And the "edge" version is where any effort for ARM is going.  You may be best off requesting better feature parity for that release.

I would also suggest considering PostgreSQL as well, depending on how difficult porting would be for you, but you may run in to similar issues with specific use cases locally.  Having gone through several painful database platform migrations in the early 00's it is emphatically not fun and I don't recommend it.

An alternative might be a cloud hosted instance with each dev working on a separate `database` or sets of databases... not as clean as a fresh docker image, but at least something to move forward.  Again, depends on your need/load.

**Comment by SuperFola** [comment]
There is the solution of running the container inside a VM, such as colima which can interface with docker pretty easily. However, even if the container spins to life for me, is reachable and all, it's nearly impossible to get liquibase to communicate with it to import databases. Half of a solution.

For those wanting to try this out, I'm using `colima start --arch amd --cpu 5 -memory 5 --dns 8.8.8.8` on a Mac M1 Pro. If you have any certificate problem when trying to pull images, this helped me getting it to work: https://stackoverflow.com/a/55260438/13157652.

**Comment by yk-jemmic** [comment]
M1 support is needed urgently!!!

**Comment by sadbuttrue1** [comment]
M1 support is needed urgently!!!

**Comment by SuperFola** [comment]
Please leave thumbs up on the relevant messages instead of polluting the thread and not bringing anything.

**Comment by Meligy** [comment]
Sorry for spamming, I just want to say that while workaround like colima exist, they have issues. Some are obvious (the degradation of performance emulating x86), some are not so (the emulation, being a hack it can break eventually, like https://github.com/abiosoft/colima/issues/288).

**Comment by JohannesHoppe** [comment]
Bumping this for visibility.

We pay a ton of money for our Azure and I expect Microsoft to fix a problem on time, only once please. But as usual, disappointment all the way.

**Comment by teodoramiteva** [comment]
M1 support is needed urgently

**Comment by darrinholst** [comment]
> @SuperFola could you please share with the example of switching Docker contexts? I want to run my docker compose with SQL server under Colima and other Docker instances as usual under ARM virtualization. Thank you

`DOCKER_CONTEXT=colima docker compose up`

It does work, but it's miserably slow. I switched to using an ec2 instance as a remote docker host.

**Comment by JohannesHoppe** [comment]
@meyntony are you joking? This issue is not resolved at all. The emulation with colima is an unbelievable slow workaround. We are all working for paying customers (I swear, I would never consider SQL server for my own projects) and Microsoft should provide a real solution!!

**Comment by JohannesHoppe** [comment]
@meyntony edit: Fixed my harsh wording...

**Comment by meyntony** [comment]
Closed by mistake while I was trying to unsubscribe. Sorry.

**Comment by JohannesHoppe** [comment]
If you're using a lot of T-SQL with a lot of data, it's definitely very slow. I'm glad it works for you on your machine with your data, but it doesn't change the fact that emulating a different processor architecture is slow as hell.

**Comment by ramonesteban78** [comment]
M1 support is needed urgently

**Comment by Dzivo** [comment]
M1 support is needed urgently

**Comment by Joel-Duff** [comment]
M1 support is needed urgently

**Comment by grishick** [comment]
+1 really need this for local development

**Comment by PraJaL55** [comment]
+1 need this

**Comment by dashkodo** [comment]
+1, sql server edge is too limited

**Comment by thisisthekap** [comment]
Docker is going to add support without QEMU: https://github.com/docker/roadmap/issues/384

**Comment by aldo1400** [comment]
M1 support is needed urgently

**Comment by MrBogomips** [comment]
Azure Sql Edge is not a viable option!

Lacking of fundamentals feature and a lot of issues with tooling.

**Please add support support for Apple M1 silicon.**

**Comment by code-dm** [comment]
M1 support is needed urgently

**Comment by ramzzzay** [comment]
2+ years past since WWDC 2020. We're all need support for Apple silicon.

**Comment by jordantshaw** [comment]
How has this not been fixed yet.  This is an absolute must for local development

**Comment by jordantshaw** [comment]
> > The emulation with colima is an unbelievable slow workaround.
> 
> I've been using this for a while on my M1 Pro and don't have any slowness problem, you must be doing something wrong. Only SQLServer needs to run inside colima, not your other containers.

Im also experiencing extreme slowness with this colima solution.  Anyway you can outline in more detail what you're doing to avoid that?

**Comment by konstantinos-kafkas-cko** [comment]
Please add support for Apple M1 silicon.

**Comment by ZachBourque** [comment]
M1 support needed urgently please :)

**Comment by botanozdemir** [comment]
THIS IS A BLOCKER ISSUE. M1 SUPPORT NEEDED URGENTLY! COLIMA AND AZURE EDGE NOT VIABLE SOLUTIONS.

**Comment by tracker1** [comment]
@botanozdemir Yelling will not draw attention any faster...

**Comment by botanozdemir** [comment]
@tracker1 just tried...

**Comment by konstantinos-kafkas-cko** [comment]
Guys we need this urgently, please follow up

**Comment by dkuhta** [comment]
M1 support needed urgently!

**Comment by yori-s** [comment]
This has come up in other comments, but I would like chime in that Azure SQL Edge was not an adequate substitute in my case,
which is making an existing dockerized development environment availble to M1 users:
* full-text search - minimally used, so stripped it out for the purposes of testing edge
* CLR support - failed over 50% of ~1200 integration tests, `AT TIME ZONE` being a major culprit
* 'Mounting a volume does not work on Docker for Mac' - aside from the inconvenience, this is my first encounter with an image that doesn't support volumes, and runs counter to the purpose of the exercise

It would be nice to get an authoritative - **yes** we'll support M1 (when) or **no**, not going to happen

**Comment by AndreaGernone** [comment]
Hi everyone! If you're still having issues with the latest image and this command: 

`docker run --platform linux/amd64 --name SQLServer -e 'ACCEPT_EULA=Y' -e 'SA_PASSWORD={YOUR_PASSWORD}*AL' -e 'MSSQL_PID=Developer' -p 1433:1433  mcr.microsoft.com/azure-sql-edge:latest`

I personally solved it by using a container instead of the run command. The reason is that they pull 2 different images and the one pulled from the yml file is the working one on M1 Mac. 

So, simply create a docker-compose.yml file with this inside: 
`
version: "3"
services:
  mssql:
    image: mcr.microsoft.com/azure-sql-edge:latest
    user: root
    environment:
      - MSSQL_PID=Developer
      - MSSQL_SA_PASSWORD={YOUR_PASSWORD}
      - ACCEPT_EULA=Y
    volumes:
      - mssql:/var/opt/mssql/data
    ports:
      - "1433:1433"
volumes:
   mssql:
   `
   
   And then use the command `docker-compose up` . It will pull the right image and start running with no problems. 
   Cheers

**Comment by tracker1** [comment]
> * 'Mounting a volume does not work on Docker for Mac' - aside from the inconvenience, this is my first encounter with an image that doesn't support volumes, and runs counter to the purpose of the exercise

Haven't had a problem mounting in a mac... I will say that mount performance is problematic in mac and windows directories, and can be slow to synchronize and may experience timing issues in practice.  Best to use a volume container and reference that.  Can still copy in/out as needed.  I do mount directly for backup directory for backup/restore command operations.

> It would be nice to get an authoritative - **yes** we'll support M1 (when) or **no**, not going to happen

I wouldn't hold my breath... honestly, may see fixes come from apple and docker for the underlying issues before a fix from MS for this.  Might be more likely to get the missing features you need added to the azure-sql-edge service.

**Comment by TheMuyu** [comment]
Apple M1 Silicon support needed urgently!

**Comment by osamahq** [comment]
M1 support is needed urgently

**Comment by samuelkitazume** [comment]
M1 support is needed urgently

**Comment by kralle333** [comment]
M1 support is needed urgently

**Comment by wcoder** [comment]
M1 support is needed urgently

**Comment by Wertros** [comment]
M1 support is needed urgently

**Comment by devanilish** [comment]
Please add support for Apple M1 silicon urgently!!

**Comment by jclausen** [comment]
Since M1 is now the only available processor arch for MacBook Pro machines, as developers upgrade their hardware, they lose the ability to use the SQL Server image.

The Azure Edge image works for basic functionality against SQL server, but does not include full text, spatial functions, nor simple conversion functions like `cast( myColumnName as "hierarchyId")`.

With the availability of Docker BuildX, I fail to see why building to support arm64 is problematic, and have a hard time believing the SQL Server for Linux team can't figure out how to handle ARM architecture to enable this feature set.  The issues around ARM support are only going to continue to magnify in the months and years to come.

**Comment by newGeneracion** [comment]
This worked for me, my dockerfile (special attention to freetds-bin freetds-dev)

`FROM python:3.10.0

ENV ENV local
ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONUNBUFFERED=1

RUN apt update -y

RUN apt install -y libxml2-dev libxmlsec1-dev libxmlsec1-openssl xmlsec1 libxslt1-dev \
  xvfb libfontconfig python3-dev build-essential wkhtmltopdf gettext freetds-bin freetds-dev

RUN pip install pip==22.3.1

RUN mkdir -p /voldemort_app/tmp

COPY ./requirements.txt /voldemort_app/requirements.txt

RUN pip install --no-cache-dir -r /voldemort_app/requirements.txt \
  --extra-index-url=https://pypi.viewflow.io/12F7DABD8D91/simple/

COPY . /voldemort_app

RUN adduser --disabled-password --gecos '' myuser

RUN chmod +x /voldemort_app/docker-compose/*/entrypoint*.sh \
  && chmod +x /voldemort_app/docker-compose/wait-for-it.sh 

WORKDIR /voldemort_app

EXPOSE 5000`

**Comment by hoho00** [comment]
M1 support is needed urgently

**Comment by tracker1** [comment]
https://github.com/docker/roadmap/issues/384

**Comment by jclausen** [comment]
> [docker/roadmap#384](https://github.com/docker/roadmap/issues/384)

I've seen that.  It's not an optimal solution as the Rosetta 2 emulation is significantly worse in performance.  The solution is just to use Docker BuildX to build multi-arch containers, [like so](https://github.com/Ortus-Solutions/docker-commandbox/blob/development/.github/workflows/release.yml#L126).  The Azure Edge containers are doing this already.

**Comment by tracker1** [comment]
> I've seen that. It's not an optimal solution as the Rosetta 2 emulation is significantly worse in performance. The solution is just to use Docker BuildX to build multi-arch containers, [like so](https://github.com/Ortus-Solutions/docker-commandbox/blob/development/.github/workflows/release.yml#L126). The Azure Edge containers are doing this already.

Where Rosetta 2 is available, it's not that much worse and significantly better than the QEMU emulation being used for x86_64 currently.

While a native aarch64 build of SQL Server would be better still, I don't know that it's even on Microsoft's radar, as a single ARM core is unlikely to work well, and the Azure Edge containers, as you mention already do this.  Asking for the missing features in Edge version are probably more likely to see response sooner than getting a full SQL Server on Arm release at all.

**Comment by thisisthekap** [comment]
I can confirm, that running SQL Server with Rosetta 2 works like a charm.

What I did to get it working using colima:

```bash
brew install --HEAD lima
git checkout support-vz
make build
sudo make install

# set driver explicitly to vz in this step
colima template

# --network-address is needed to have data transfer to containers working as expected
# number of cpus and amount of GBs RAM may be defined according to your own needs
colima start --network-address --cpu 8 --memory 8
```

This script is based of the work of @mmeyer724 (https://github.com/docker/roadmap/issues/384#issuecomment-1329712712).

Make sure to run `docker context list` to check which docker context is used (the new colima context needs to be selected).

After that, you may just run `docker run -e ACCEPT_EULA=Y mcr.microsoft.com/mssql/server:2019-CU18-ubuntu-20.04`

**Comment by Will-Bill** [comment]
Above is slightly incorrect, need to clone colima. 

```
brew install --HEAD lima

git clone https://github.com/abiosoft/colima.git && cd colima
git checkout support-vz
make build
sudo make install

colima start
```

**Comment by adam-at-mobomo** [comment]
Thanks, @thisisthekap and @Will-Bill.  The above almost worked for me, but I am not trying to run SQL Server.  Instead, I need to run a container and also do a volume mount, and that did not work.  However, if you make one more change, it does:

```
# set driver explicitly to vz and also set the mountType to virtiofs in this step
colima template
```

**Comment by invaderb** [comment]
Tries the steps above and is works on a m1 pro with colima but on a m1 max it returns a warning and nothing actually loads
[The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64)](https://stackoverflow.com/questions/71000707/docker-get-started-warning-the-requested-images-platform-linux-arm64-v8-doe)

**Comment by adam-at-mobomo** [comment]
> Tries the steps above and is works on a m1 pro with colima but on a m1 max it returns a warning and nothing actually loads
> [The requested image's platform (linux/arm64/v8) does not match the detected host platform (linux/amd64)](https://stackoverflow.com/questions/71000707/docker-get-started-warning-the-requested-images-platform-linux-arm64-v8-doe)

@invaderb, I'm not sure why Max would be any different than Pro.  I assume you are getting this when trying to run a docker image.  Try adding `--platform linux/amd64` like this:

```
docker run --rm --platform linux/amd64 busybox arch
```

(for whatever x86 image you are trying to run)

**Comment by invaderb** [comment]
@adam-at-mobomo Yeah I was adding it to the docker file FROM --platform=linux/amd64 php:7.3-apache-buster but I needed to add it to the docker compose yml file as well which resolved the warning but having some other volume sync issue now but sure it's not related to this.

**Comment by overengineeredit** [comment]
I tried the update to use colima with container mcr.microsoft.com/mssql/server:2017-latest to no effect. Am still getting the same error as the initially noted one:

```/opt/mssql/bin/sqlservr: Invalid mapping of address 0x40094e5000 in reserved address space below 0x400000000000. Possible causes:
1) the process (itself, or via a wrapper) starts-up its own running environment sets the stack size limit to unlimited via syscall setrlimit(2);
2) the process (itself, or via a wrapper) adjusts its own execution domain and flag the system its legacy personality via syscall personality(2);
3) sysadmin deliberately sets the system to run on legacy VA layout mode by adjusting a sysctl knob vm.legacy_va_layout.```

**Comment by invaderb** [comment]
> I tried the [suggest steps to use colima](https://github.com/microsoft/mssql-docker/issues/734#issuecomment-1333696500)  with container mcr.microsoft.com/mssql/server:2017-latest to no effect. Am still getting the same error as the initially noted one:
> 
> 
> 
> ```
> 
> /opt/mssql/bin/sqlservr: Invalid mapping of address 0x40094e5000 in reserved address space below 0x400000000000. Possible causes:
> 
> 1) the process (itself, or via a wrapper) starts-up its own running environment sets the stack size limit to unlimited via syscall setrlimit(2);
> 
> 2) the process (itself, or via a wrapper) adjusts its own execution domain and flag the system its legacy personality via syscall personality(2);
> 
> 3) sysadmin deliberately sets the system to run on legacy VA layout mode by adjusting a sysctl knob vm.legacy_va_layout.```

are you sure your using the Colima runtime?

run "docker context list" and see what it's set too

**Comment by sinfergit** [comment]
> > I tried the [suggest steps to use colima](https://github.com/microsoft/mssql-docker/issues/734#issuecomment-1333696500)  with container mcr.microsoft.com/mssql/server:2017-latest to no effect. Am still getting the same error as the initially noted one:
> > ```
> > 
> > /opt/mssql/bin/sqlservr: Invalid mapping of address 0x40094e5000 in reserved address space below 0x400000000000. Possible causes:
> > 
> > 1) the process (itself, or via a wrapper) starts-up its own running environment sets the stack size limit to unlimited via syscall setrlimit(2);
> > 
> > 2) the process (itself, or via a wrapper) adjusts its own execution domain and flag the system its legacy personality via syscall personality(2);
> > 
> > 3) sysadmin deliberately sets the system to run on legacy VA layout mode by adjusting a sysctl knob vm.legacy_va_layout.```
> > ```
> 
> are you sure your using the Colima runtime?
> 
> run "docker context list" and see what it's set too

I got the same issue and checked the docker context as in the below image. Colima has been selected as it should be.
<img width="1190" alt="Screenshot 2022-12-30 at 6 13 49 PM" src="https://user-images.githubusercontent.com/24528193/210059922-f90ac2f5-ae45-44e2-81f8-ac5fe6f86cba.png">

Any idea as to what could be the reason?

**Comment by darrinholst** [comment]
> > > I tried the [suggest steps to use colima](https://github.com/microsoft/mssql-docker/issues/734#issuecomment-1333696500)  with container mcr.microsoft.com/mssql/server:2017-latest to no effect. Am still getting the same error as the initially noted one:
> > > ```
> > > 
> > > /opt/mssql/bin/sqlservr: Invalid mapping of address 0x40094e5000 in reserved address space below 0x400000000000. Possible causes:
> > > 
> > > 1) the process (itself, or via a wrapper) starts-up its own running environment sets the stack size limit to unlimited via syscall setrlimit(2);
> > > 
> > > 2) the process (itself, or via a wrapper) adjusts its own execution domain and flag the system its legacy personality via syscall personality(2);
> > > 
> > > 3) sysadmin deliberately sets the system to run on legacy VA layout mode by adjusting a sysctl knob vm.legacy_va_layout.```
> > > ```
> > 
> > 
> > are you sure your using the Colima runtime?
> > run "docker context list" and see what it's set too
> 
> I got the same issue and checked the docker context as in the below image. Colima has been selected as it should be. <img alt="Screenshot 2022-12-30 at 6 13 49 PM" width="1190" src="https://user-images.githubusercontent.com/24528193/210059922-f90ac2f5-ae45-44e2-81f8-ac5fe6f86cba.png">
> 
> Any idea as to what could be the reason?

Run `colima status` to make sure it’s using the new virtualization framework and not qemu still.

**Comment by sinfergit** [comment]
> > > > I tried the [suggest steps to use colima](https://github.com/microsoft/mssql-docker/issues/734#issuecomment-1333696500)  with container mcr.microsoft.com/mssql/server:2017-latest to no effect. Am still getting the same error as the initially noted one:
> > > > ```
> > > > 
> > > > /opt/mssql/bin/sqlservr: Invalid mapping of address 0x40094e5000 in reserved address space below 0x400000000000. Possible causes:
> > > > 
> > > > 1) the process (itself, or via a wrapper) starts-up its own running environment sets the stack size limit to unlimited via syscall setrlimit(2);
> > > > 
> > > > 2) the process (itself, or via a wrapper) adjusts its own execution domain and flag the system its legacy personality via syscall personality(2);
> > > > 
> > > > 3) sysadmin deliberately sets the system to run on legacy VA layout mode by adjusting a sysctl knob vm.legacy_va_layout.```
> > > > ```
> > > 
> > > 
> > > are you sure your using the Colima runtime?
> > > run "docker context list" and see what it's set too
> > 
> > 
> > I got the same issue and checked the docker context as in the below image. Colima has been selected as it should be. <img alt="Screenshot 2022-12-30 at 6 13 49 PM" width="1190" src="https://user-images.githubusercontent.com/24528193/210059922-f90ac2f5-ae45-44e2-81f8-ac5fe6f86cba.png">
> > Any idea as to what could be the reason?
> 
> Run `colima status` to make sure it’s using the new virtualization framework and not qemu still.

When I ran `colima status` it gives me an error saying  `FATA[0000] colima is not running ` but when I run the command to start Colima it runs without any issue. 
Also I noticed that vmType in Colima Template was as qemu and I manually changed it to vz but still the same error occurs

**Comment by invaderb** [comment]
@sinfergit @darrinholst did you get any errors when you installed it? I'd suggest clearing everything out and re-installing lima and colima by following the steps above it sounds like something is borked.

**Comment by sinfergit** [comment]
> @sinfergit @darrinholst did you get any errors when you installed it? I'd suggest clearing everything out and re-installing lima and colima by following the steps above it sounds like something is borked.

The only error I got running the script is from here `error: pathspec 'support-vz' did not match any file(s) known to git`
And when I checked there is no such branch in Colima Git repository and I checked out to main branch.

**Comment by darrinholst** [comment]
> > @sinfergit @darrinholst did you get any errors when you installed it? I'd suggest clearing everything out and re-installing lima and colima by following the steps above it sounds like something is borked.
> 
> The only error I got running the script is from here `error: pathspec 'support-vz' did not match any file(s) known to git` And when I checked there is no such branch in Colima Git repository and I checked out to main branch.
> 
> Did everything from the scratch again and now getting the following error about the colima driver. <img alt="Screenshot 2022-12-31 at 4 28 54 PM" width="608" src="https://user-images.githubusercontent.com/24528193/210130539-4dc11ca7-21a1-4b20-a9e8-6d7e43647d4f.png">

Try `colima delete` first before starting it. Looks like it's using the previous VM.

Also you can use `brew install --HEAD colima` instead of building it manually since that branch has been merged to main.

**Comment by darrinholst** [comment]
You shouldn't have to build anything from source anymore. Here is what I just did to clean and reinstall...

    brew uninstall --force colima lima
    rm -rf ~/.colima ~/.lima
    brew install colima
    colima template

Change `vmType` to `vz` and `mountType` to `virtiofs`. I also adjust CPU, disk, and memory here so I don't have to add them to the start command.

    colima start
    colima status

```
INFO[0000] colima is running using macOS Virtualization.Framework
INFO[0000] arch: aarch64
INFO[0000] runtime: docker
INFO[0000] mountType: virtiofs
INFO[0000] socket: unix:///Users/darrinholst/.colima/default/docker.sock
```

    docker run --rm -e ACCEPT_EULA=Y mcr.microsoft.com/mssql/server:2019-CU18-ubuntu-20.04

```
2022-12-31 13:47:53.57 spid24s     Server is listening on [ 'any' <ipv6> 1433].
2022-12-31 13:47:53.58 spid24s     Server is listening on [ 'any' <ipv4> 1433].
2022-12-31 13:47:53.59 Server      Server is listening on [ ::1 <ipv6> 1434].
2022-12-31 13:47:53.59 Server      Server is listening on [ 127.0.0.1 <ipv4> 1434].
2022-12-31 13:47:53.59 Server      Dedicated admin connection support was established for listening locally on port 1434.
2022-12-31 13:47:53.59 spid24s     Server is listening on [ ::1 <ipv6> 1431].
2022-12-31 13:47:53.59 spid24s     Server is listening on [ 127.0.0.1 <ipv4> 1431].
2022-12-31 13:47:53.60 spid24s     SQL Server is now ready for client connections. This is an informational message; no user action is required.
```

**Comment by sinfergit** [comment]
> You shouldn't have to build anything from source anymore. Here is what I just did to clean and reinstall...
> 
> ```
> brew uninstall --force colima lima
> rm -rf ~/.colima ~/.lima
> brew install colima
> colima template
> ```
> 
> Change `vmType` to `vz` and `mountType` to `virtiofs`. I also adjust CPU, disk, and memory here so I don't have to add them to the start command.
> 
> ```
> colima start
> colima status
> ```
> 
>  ```
> INFO[0000] colima is running using macOS Virtualization.Framework
> INFO[0000] arch: aarch64
> INFO[0000] runtime: docker
> INFO[0000] mountType: virtiofs
> INFO[0000] socket: unix:///Users/darrinholst/.colima/default/docker.sock
> ```
> 
>  ```
> docker run --rm -e ACCEPT_EULA=Y mcr.microsoft.com/mssql/server:2019-CU18-ubuntu-20.04
> ```
> 
> ```
> 2022-12-31 13:47:53.57 spid24s     Server is listening on [ 'any' <ipv6> 1433].
> 2022-12-31 13:47:53.58 spid24s     Server is listening on [ 'any' <ipv4> 1433].
> 2022-12-31 13:47:53.59 Server      Server is listening on [ ::1 <ipv6> 1434].
> 2022-12-31 13:47:53.59 Server      Server is listening on [ 127.0.0.1 <ipv4> 1434].
> 2022-12-31 13:47:53.59 Server      Dedicated admin connection support was established for listening locally on port 1434.
> 2022-12-31 13:47:53.59 spid24s     Server is listening on [ ::1 <ipv6> 1431].
> 2022-12-31 13:47:53.59 spid24s     Server is listening on [ 127.0.0.1 <ipv4> 1431].
> 2022-12-31 13:47:53.60 spid24s     SQL Server is now ready for client connections. This is an informational message; no user action is required.
> ```

Thanks a lot @darrinholst now it works. So what was missing was the mountType to be updated to vz.
Thanks for you time 👍

**Comment by Meligy** [comment]
What's the point of changing Colima defaults for SQL Server? I have been running it successfully for a year without changing anything.

Is it faster or anything?

I also don't share volumes with the host. Is that what's broken by the defaults?

**Comment by invaderb** [comment]
> What's the point of changing Colima defaults for SQL Server? I have been running it successfully for a year without changing anything.
> 
> 
> 
> Is it faster or anything?
> 
> 
> 
> I also don't share volumes with the host. Is that what's broken by the defaults?

@Meligy the issue specifically steams for m1 macs as there isn't a native mssql container. if your on an Intel Mac or windows there isn't a strong reason to switch.

**Comment by Meligy** [comment]
> @Meligy the issue specifically steams for m1 macs as there isn't a native mssql container

@invaderb, yes. I get that. I have an M1 Max MacbBook Pro 16". And I use colima for that.

Sorry if my question was not clear but I was asking about the reason for customising the colima template vs using colima's defaults, for someone who's already using colima to run SQL Server on Mac.

**Comment by darrinholst** [comment]
> Is it faster or anything?

Yes. Significantly. For some people just getting it to run with qemu emulation on arm64 was good enough. For others the performance was a show stopper.

In my case we’re using it for automated testing against a local database (with the need for the CLR so the azure image was out). It’s the difference between seconds and literally minutes.

**Comment by Meligy** [comment]
OK. The new settings in colima ARE cool.

I forgot to measure the colima start command, it certainly felt faster, but I could see the speeds of starting a custom SQL Server image (for enabling Full Text Search etc.) affected a lot.

I applied the settings for colima template:

```yaml
vmType: vz
mountType: virtiofs
```

Then I ran `colima delete`.

I changed my start command

From (OLD):
```pwsh
colima start --arch x86_64 --cpu-type kvm64 --cpu 4 --memory 4 --disk 60
```

To (NEW):
```pwsh
colima start --network-address --cpu 4 --memory 4 --disk 60
```

Looking at the defaults in the template, I think at least setting `cpu-type` was reverting to QEMU.

And the results were amazing.

Results BEFORE (I cancelled when I sensed nothing changed from usual):
![image](https://user-images.githubusercontent.com/171637/210175113-81a1c0dc-24bb-4978-b61d-06b1cefa3d9b.png)

Results AFTER (Look at same-shape arrows for comparison):
![image](https://user-images.githubusercontent.com/171637/210175122-9d19d9ff-4d75-44b8-a2fe-31a256ea0e49.png)

The download time of the 784.49MB part ending in `f3a3464f` went down from 64.3s to 48.8s, but the most amazing time difference was the extraction of the same part, going down from 211.2s to just 12.4s!! Not to mention that also my custom `apt-get` commands time went from 142.0s to 45.6s.

That's going from minutes to seconds as @darrinholst said!

Thanks for sharing!

And hopefully at some point people won't have to go through all of this.

**Comment by jordantshaw** [comment]
UPDATE: The new docker-desktop release (v4.16.0) introduces a new Mac beta feature, Rosetta for Linux, which helps performance and stability when running intel bases images.  I am now able to run MSSQL docker image on my Mac M1 laptop with no issues yet.  Also seems pretty performant, but have not run any benchmarks yet.

Release Notes:
[New Beta feature for MacOS 13, Rosetta for Linux, has been added for faster emulation of Intel-based images on Apple Silicon.](https://docs.docker.com/desktop/release-notes/#4160)

**Comment by dzsquared** [comment]
Recapping where arm64 (M1/M2) support is currently for SQL Server:
- to run the mssql container, enable Rosetta for emulation in Docker Desktop v4.16 or higher https://devblogs.microsoft.com/azure-sql/development-with-sql-in-containers-on-macos/
  - alternatively, you can setup Colima
  - also alternatively, you can use Azure SQL Edge
- Azure Data Studio has an arm64 macOS build https://github.com/microsoft/azuredatastudio
  - alternatively, you can use the mssql extension in VS Code
- go-sqlcmd cli has native arm64 builds available in brew and for direct download https://docs.microsoft.com/sql/tools/go-sqlcmd-utility

if I've missed anything from the 91 comments in this thread, please feel free to open a new issue on that specific item!

**Comment by glerchundi** [comment]
@dzsquared thanks for the list! Would you mind adding to that list that macOS Ventura (13) is required? I've gone crazy looking for that checkbox within the Docker Desktop and realised that it wasn't present because I was on 12.

**Comment by Meligy** [comment]
I don't know about the need for latest Mac, but there's another gotcha to be aware of that I mentioned here: https://github.com/docker/roadmap/issues/384#issuecomment-1382615753.

In short: there are 2 beta features in the same tab, the Rosetta emulation and containerd for image pulling and storage. The containerd feature is supposed to be much faster but in my experience it caused an error, and after turning it off, the image pulling and extraction was still fairly fast (71.7s vs 61.3s on colima [as mentioned above](https://github.com/microsoft/mssql-docker/issues/734#issuecomment-1368462957)).

**Comment by Jolg42** [comment]
Someone confirmed macOS Ventura is required, see https://github.com/docker/roadmap/issues/384#issuecomment-1385037769, I'll upgrade soon to get this on my computer.

**Comment by tinydogio-joshua** [comment]
Are there any plans for an official ARM container? My understanding is that Rosetta will most likely go away at some point. It would be nice to see an official ARM build.

**Comment by ramzzzay** [comment]
+1 for native arm build, because Rosetta 2 like Rosetta 1 -- just temporary workaround for 2-4 years.

**Comment by mkurz** [comment]
Yeah, that would be really great! aarch64 build now :wink:

**Comment by jclausen** [comment]
I can confirm that running the x86_64 SQL Server container under Rosetta is not an ideal solution.  Due to the arch emulation, the SQL Server container under load locks up and becomes unresponsive. My M1-using devs report they are constantly having to restart the SQL Server container.  They can use it, but it's buggy.

**Comment by mkurz** [comment]
Yeah it sucks! Give us the real thing :pray:

**Comment by dzsquared** [comment]
Alright folks - there's a new issue to track for a native aarch container - you're welcome to bring your concrete feedback to the comments - https://github.com/microsoft/mssql-docker/issues/802
